### DISCUSSION TITLE
does kloadgen value config support replacement of variables set through beanshell etc.

### DISCUSSION THREAD

**Discussion by pdeole** [sngular/kloadgen#133] [Q&A] [answered]
Hi,

I would like to know following: 
our requirement is to generate load on kafka topics for predefined avro schemas. For that we would like to try using kloadgen plugin.
However while generating load, we want the parameter values for avro schema to be set dynamically for which we want to use beanshell to generate variable using some logic and then use that variable to set in the value schema. 
Is this supported in kloadgen?

**Comment by Rivarsal** [comment, marked as answer]
Hello @pdeole 

Yes, it is supported by Kloadgen.

**Reply by pdeole** [reply]
Thanks @Rivarsal  would you be able to provide me a sample jmx that i can refer for this? e.g. how do i also use a user defined variable in the Value file serialized config or even a variable set through beanshell, how can be used in value file config?

**Reply by pdeole** [reply]
@Rivarsal @jemacineiras  i checked the Example-Variable-Substitution.jmx script in the source and i see it is using jsr223 preprocessor to set a variable dynamically, however i did not find any reference where that variable being used in config element. Is the script there outdated? If so, then can you pls get the latest version uploaded so we can refer that to develop our performance test scripts.

**Reply by Rivarsal** [reply]
Hello @pdeole!

Sorry for the late answer, you have to type ${valueToReplace} in the FieldValueList box where you want the variable to be assigned in order to make that test work. Also be wary that you need to change the **Name Strategy** to **net.coru.kloadgen.strategy.RecordSubjectNameStrategy** and the **Serializer** to **net.coru.kloadgen.serializer.AvroSerializer** in order to send the schema to Kafka.

Cheers!
